celer/homotopy.py

```diff
@@ -185,14 +196,16 @@ def celer_path(X, y, pb, eps=1e-3, n_alphas=100, alphas=None,
     X_dense, X_data, X_indices, X_indptr = _sparse_and_dense(X)
 
     weights = _check_weights(weights, pb, X, n_groups)
+    # to prevent ref before assignment in dnorm_enet
+    w = np.zeros(n_features, dtype=X.dtype)
```

Comment:
if you use w = 0 then it's like using l1_ratio=1 no ? Then you may consider making w an optional parameter in dnorm_enet to avoid this.

Reply:
@mathurinm, there was a slight confusion in this case. Indeed, I will need to initialize ``w`` here only in the case where ``alphas`` are not provided. In this case, ``w = 0`` or ``l1_ratio = 1`` all are the same.

Reply:
Also, I didn't think about passing ``w`` as optional params to avoid initializing a ``w = 0`` every time the ``dnorm_enet`` is called